### DISCUSSION TITLE
What is a typical CI/CD Workflow for developing Channels

### DISCUSSION THREAD

**Discussion by GrantMilesNHS** [nextgenhealthcare/connect#5214] [Q&A]
My end goal here is to have all exported channels source controlled alongside the rest of our code base. My first thought was to have:
1. Create a channel and test in a local instance of Connect (in a docker container on dev machine)
2. Export that channel and save it in source
3. Use Azure DevOps Pipelines^ and Connect API's to create/update channels in UAT while transforming things like username, password, connection string etc. in the pipeline
4.  PR trigger the prod build and apply the prod transform to the same XML file that is in source

^ Pipeline template checks if the channel exists (HTTP 200/204) and conditionally creates or updates dependant on response then deploys if all tasks are successful.

Our discussions suggest that this is consistent with current processes and plausible however the sticking point at the moment is how we would get around the transformation of the XML document before including it on the POST/PUT

This is a fairly standard practice and maybe we need to utilise Mirth Connect better for this kind of release management and I am open to creative solutions. During the first version of our pipeline the Create Channel works well but the deploy doesnt work because we are unable to XML transform the channel ID (we haven't yet looked at further configurations)

So, What is a typical CI/CD Workflow for developing Channels? How do people manage the releasing of new channels and updating new ones?

**Comment by jonbartels** [comment]
https://github.com/nextgenhealthcare/connect/discussions/4689 may have some useful information. Briefly, there is no single tool or one way to do this with MC. I'm confident the tools and pieces are there but no one has openly put those pieces together and published a standard CI/CD pipeline for MC.
